turbopack/crates/turbopack-core/src/diagnostics/mod.rs

```diff
@@ -101,7 +101,14 @@ where
 {
     async fn peek_diagnostics(self) -> Result<CapturedDiagnostics> {
         Ok(CapturedDiagnostics {
-            diagnostics: self.peek_collectibles(),
+            diagnostics: self
+                .peek_collectibles()
+                .into_iter()
+                .map(|v: Vc<Box<dyn Diagnostic>>| v.to_resolved())
+                .try_join()
+                .await?
+                .into_iter()
+                .collect(),
```

Comment:
There are multiple changes where we are now doing two collects in a row. Is there some way around that?

Reply:
It's impossible because `try_join()` always returns `Vec<T>`